test/models/channel_test.rb

```diff
@@ -94,6 +94,29 @@ class ChannelTest < ActionDispatch::IntegrationTest
     assert_equal "can't be changed", channel.errors.messages[:details][0]
   end
 
+  test "public_name must be unique" do
+    channel = channels(:google_verified)
+    channel.public_name = "test_value"
+    channel.save!
+
+    channel_2 = channels(:new_site)
+    channel_2.public_name = channel.public_name
+    refute channel_2.valid?
+
+    assert_equal "has already been taken", channel_2.errors.messages[:public_name][0]
+  end
+
+  # test "public_identifier is added on create and must be unique" do
```

Comment:
Comments

Reply:
Same issue as above- once the rake task has been run, I can uncomment the validation and the test will pass.